### PR TITLE
Add kwarg to not check spatial consistency

### PR DESCRIPTION
**Pull request by fepegar**
Related to #734.

**Description**
The new kwarg allows to avoid checks for spatial consistency across images within a subject. This is useful when we are certain that the images are in the same physical space but different voxel space or different orientation. See the @Spenhouet's issue for more information:
- #734

The recommended usage in this case is

```python
RandomAffine(center='origin', check_shape=False)
```

**Checklist**

- [x] I have read the [`CONTRIBUTING`](https://github.com/fepegar/torchio/blob/main/CONTRIBUTING.rst) docs and have a developer setup (especially important are `pre-commit`and `pytest`)
- [x] Non-breaking change (would not break existing functionality)
- [ ] Breaking change (would cause existing functionality to change)
- [x] Tests added or modified to cover the changes
- [x] Integration tests passed locally by running `pytest`
- [x] In-line docstrings updated
- [x] Documentation updated, tested running `make html` inside the `docs/` folder
- [x] This pull request is ready to be reviewed
- [ ] If the PR is ready and there are multiple commits, I have [squashed them and force-pushed](https://www.w3docs.com/snippets/git/how-to-combine-multiple-commits-into-one-with-3-steps.html#force-pushing-commits-7)

### CODE REVIEW

**Comment by fepegar** [comment]
@Spenhouet could you please take a look? I used this code on your data and it worked fine:

```python
from pathlib import Path
import torchio as tio

ims_dir = Path('/tmp/example_images')
subject = tio.Subject(
    im=tio.ScalarImage(ims_dir / 't1.nii.gz'),
    lm=tio.LabelMap(ims_dir / 'thalamic_reorient.nrrd'),
)
transform = tio.RandomAffine(center='origin', check_shape=False)
transformed_subject = transform(subject)
transformed_subject.im.save('/tmp/t1_tr.nii.gz')
transformed_subject.lm.save('/tmp/t1_thalamus_seg.nii.gz')
```

`thalamic_reorient.nrrd` is the thalamus label map reoriented into some random orientation, to make sure the transform is robust to different orientations.

**Comment by Spenhouet** [comment]
@fepegar I tested it with more labels with different resolution and shape.
It works nicely! Thanks a lot!

**Comment by fepegar** [comment]
No worries! Do you mind if I add you to the list of contributors?

**Comment by fepegar** [comment]
I'll take the upvote as a yes!

@all-contributors please add @Spenhouet for ideas
